adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -129,6 +130,24 @@ AuthenticationResult acquireTokenWithRefreshToken(final String refreshToken)
                 result.setRefreshToken(refreshToken);
             }
         } catch (final IOException | AuthenticationException exc) {
+            //if get 503,504,500 && outageMode is on, return the stale token
+            if (exc instanceof AuthenticationException
```

Comment:
I could do the individually catching. However, here I am trying to modify the previous code as little as possible.